clang/test/CodeGenSYCL/anonymous_integration_footer.cpp

```diff
@@ -0,0 +1,363 @@
+// CHECK-NEXT: inline const char *get_spec_constant_symbolic_ID<::c>() {
+// CHECK-NEXT: return "";
+// CHECK-NEXT: }
+// CHECK-NEXT: } // namespace detail
+// CHECK-NEXT: } // namespace sycl
+static constexpr sycl::specialization_id d{4};
+// CHECK-NEXT: namespace sycl {
+// CHECK-NEXT: namespace detail {
+// CHECK-NEXT: template<>
+// CHECK-NEXT: inline const char *get_spec_constant_symbolic_ID<::d>() {
+// CHECK-NEXT: return "";
+// CHECK-NEXT: }
+// CHECK-NEXT: } // namespace detail
+// CHECK-NEXT: } // namespace sycl
+
+namespace {
+struct S2 {
+  static constexpr sycl::specialization_id a{18};
+  // CHECK-NEXT: namespace {
+  // CHECK-NEXT: namespace __sycl_detail {
+  // CHECK-NEXT: static constexpr decltype(S2::a) &__spec_id_shim_[[SHIM_ID:[0-9]+]]() {
+  // CHECK-NEXT: return S2::a;
+  // CHECK-NEXT: }
+  // CHECK-NEXT: } // namespace __sycl_detail
+  // CHECK-NEXT: } // namespace
+  // CHECK-NEXT: namespace sycl {
+  // CHECK-NEXT: namespace detail {
+  // CHECK-NEXT: template<>
+  // CHECK-NEXT: inline const char *get_spec_constant_symbolic_ID<::__sycl_detail::__spec_id_shim_[[SHIM_ID]]()>() {
+  // CHECK-NEXT: return ::__sycl_detail::__spec_id_shim_[[SHIM_ID]]();
+  // CHECK-NEXT: }
+  // CHECK-NEXT: // namespace detail
+  // CHECK-NEXT: // namespace sycl
+};
+} // namespace
+
+template <int Val>
+struct S3 {
+  static constexpr sycl::specialization_id a{Val};
+};
+template class S3<1>;
+// CHECK-NEXT: namespace sycl {
+// CHECK-NEXT: namespace detail {
+// CHECK-NEXT: template<>
+// CHECK-NEXT: inline const char *get_spec_constant_symbolic_ID<::S3<1>::a>() {
+// CHECK-NEXT: return "";
+// CHECK-NEXT: }
+// CHECK-NEXT: } // namespace detail
+// CHECK-NEXT: } // namespace sycl
+template class S3<2>;
+// CHECK-NEXT: namespace sycl {
+// CHECK-NEXT: namespace detail {
+// CHECK-NEXT: template<>
+// CHECK-NEXT: inline const char *get_spec_constant_symbolic_ID<::S3<2>::a>() {
+// CHECK-NEXT: return "";
+// CHECK-NEXT: }
+// CHECK-NEXT: } // namespace detail
+// CHECK-NEXT: } // namespace sycl
+
+namespace inner {
+constexpr sycl::specialization_id same_name{5};
+// CHECK-NEXT: namespace sycl {
+// CHECK-NEXT: namespace detail {
+// CHECK-NEXT: template<>
+// CHECK-NEXT: inline const char *get_spec_constant_symbolic_ID<::inner::same_name>() {
+// CHECK-NEXT: return "";
+// CHECK-NEXT: }
+// CHECK-NEXT: } // namespace detail
+// CHECK-NEXT: } // namespace sycl
+}
+constexpr sycl::specialization_id same_name{6};
+// CHECK-NEXT: namespace sycl {
+// CHECK-NEXT: namespace detail {
+// CHECK-NEXT: template<>
+// CHECK-NEXT: inline const char *get_spec_constant_symbolic_ID<::same_name>() {
+// CHECK-NEXT: return "";
+// CHECK-NEXT: }
+// CHECK-NEXT: } // namespace detail
+// CHECK-NEXT: } // namespace sycl
+namespace {
+constexpr sycl::specialization_id same_name{7};
+// CHECK-NEXT: namespace {
+// CHECK-NEXT: namespace __sycl_detail {
+// CHECK-NEXT: static constexpr decltype(same_name) &__spec_id_shim_[[SHIM_ID:[0-9]+]]() {
```

Comment:
Is it worth it to test that this SHIM_ID and others below are a different value than the previous ones?

Reply:
I'd thought about that, but I'm not aware of a good way to make them all be wildcards, yet still make sure they are different.  Unfortunately there doesn't seem to be any sort of mechanism in CHECK to do something like that.

Do you have a suggestion?

Reply:
Sorry, no, I don't have a suggestion.   It was just a thought.